_posts/2020-09-29-aws-developer-tools-mule-cicd-solution.md

```diff
@@ -0,0 +1,196 @@
+---
+layout: post
+title: Using AWS CodePipeline as CI/CD solution for MuleSoft Apps
+author: atteheino
+excerpt: Thoughts on using AWS Developer Tools to replace Jenkins as the CI/CD solution for MuleSoft Apps to be deployd to CloudHub
+tags:
+ - AWS CodeCommit
+ - AWS CodePipeline
+ - AWS CodeBuild
+ - MuleSoft
+ - Jenkins
+ - CloudHub
+ - CI/CD
+---
+
+A typical tooling setup for a integration project consists of a Git repository somewhere, Jenkins on some host, Application Server either on-premise or hosted somewhere. Git stores the source code, Jenkins builds and deploys the code to the application server running the application. 
+
+I was given the option to try out AWS alternative to this setup. AWS has [CodeCommit](https://aws.amazon.com/codecommit/) for source control, [CodeBuild](https://aws.amazon.com/codebuild/) for building your software and [CodePipeline](https://aws.amazon.com/codepipeline/) for automating the whole CI/CD process.
+
+Typically it would be a good idea to document your infrastructure as code, for example by creating the services you need via CloudFormation templates. Here I was doing a Proof Of Concept, so I did all of the setup via the AWS Console UI.
+
+# CodeCommit to store the source code
+
+It's not a requirement to use AWS CodeCommit, to be able to utilize CodeBuild or CodePipeline, but in my case it made sense as the original source code was not in a public repository like GitHub.
+
+It's simple to create a new repository to CodeCommit. All of the repositories are Git-based.
+I needed to migrate an existing repository to CodeCommit and I did it by following [these simple instructions](https://docs.aws.amazon.com/codecommit/latest/userguide/how-to-migrate-repository-existing.html).
+
+To be able to access your new repository:
+- Create a IAM user for CodeCommit access. Grant it a suitable policy to work with CodeCommit.
+- Create HTTPS Git Credentials for Git to access the CodeCommit repository. (You can do this from IAM user's Security Credentials tab)
+- Configure Git on your computer to access CodeCommit, by using the credentials created on the previous step.
+
+The rest is pretty much trivial by following the instructions on the guide.
+
+# CodePipeline to run the entire process
+
+After having the sources in CodeCommit, I took a look at the documentation on how to proceed. I figured out, that I would first create the CodePipeline that is triggered when ever the sources change in the Git repository. Second step would be to create the CodeBuild step and it's simple to add while building the CodePipeline. The CodePipeline I created is really simple:
+
+![CodePipeline example](/img/aws-developer-tools-mule-cicd-solution/pipeline.png)
```

Comment:
Image is missing but is it just not working in GitHub preview and is actually fine...?

Reply:
It works in the Docker preview, so I think it's ok...